### PR TITLE
[YUNIKORN-438] Wait for sync after apiFactory start

### PR DESCRIPTION
**Pull request by kingamarton**

### CODE REVIEW

File: pkg/client/apifactory.go [hunk 1]
```diff
@@ -203,6 +206,10 @@ func (s *APIFactory) Start() {
 	// launch clients
 	if !s.IsTestingMode() {
 		s.clients.Run(s.stopChan)
+		if err := s.WaitForSync(); err != nil {
```
Comment (wilfred-s) [inline pkg/client/apifactory.go:209]: We already know we're not in test mode why are we not calling `s.clients.WaitForSync(time.Second, 30*time.Second)` directly?

**Review by wilfred-s — COMMENTED** [review]
The other point to make (not really part of this change) we wrap returning a boolean value in a function without locking `IsTestingMode()` and use it internally. That seems just overhead for nothing inside the factory.
I can understand that from an external point of view but not for internal use.

File: pkg/client/apifactory.go (hunk 1 above)
Comment (kingamarton) [reply pkg/client/apifactory.go]: Changed it.

**Review by wilfred-s — APPROVED** [review]
change looks good

**Pull request closed without merging**